### DISCUSSION TITLE
Assertion failed when trying to `destroy` a rigid entity

### DISCUSSION THREAD

**Discussion by frutas-fruit** [xissburg/edyn#104] [General]

**Comment by xissburg** [comment]
I have pushed a fix to `master`. Thanks for reporting the issue.

**Comment by frutas-fruit** [comment]


**Reply by xissburg** [reply]
It is failing to remove a `edyn::shape_index` component, which makes it look like an internal inconsistency in Edyn, but it's hard to tell where...

**Comment by frutas-fruit** [comment]


**Reply by xissburg** [reply]
There are situations in edyn-testbed where external entities are deleted and it works fine. I wonder if this is another situation where you're deleting it right after construction, i.e. before things get setup completely.
